app/serializers/alchemy/json_api/page_serializer.rb

```diff
@@ -33,15 +33,21 @@ class PageSerializer < BaseSerializer
 
       # All public elements of this page regardless of if they are fixed or nested.
       # Used for eager loading and should be used as the +include+ parameter of your query
-      has_many :all_elements, record_type: :element, serializer: ELEMENT_SERIALIZER
+      has_many :all_elements, record_type: :element, serializer: ELEMENT_SERIALIZER do |record|
+        record.public_version.element_repository.visible
```

Comment:
I do not think we can necessarily always use the public version here.

Reply:
No, we can't. reverted.